lib/Reader/readerir.cpp

```diff
@@ -539,6 +557,19 @@ void GenIR::insertIRForSecurityObject() {
   // via the GC encoding.
 }
 
+void GenIR::callMonitorHelper(bool IsEnter) {
+  CorInfoHelpFunc HelperId;
+  const uint32_t MethodFlags = getCurrentMethodAttribs();
+  if (MethodFlags & CORINFO_FLG_STATIC) {
+    HelperId =
+        IsEnter ? CORINFO_HELP_MON_ENTER_STATIC : CORINFO_HELP_MON_EXIT_STATIC;
+  } else {
+    HelperId = IsEnter ? CORINFO_HELP_MON_ENTER : CORINFO_HELP_MON_EXIT;
+  }
+  callHelperImpl(HelperId, Type::getVoidTy(*JitContext->LLVMContext),
```

Comment:
What does arbitrary mean in this context?  Is this possibly a non-CoreCLR/modern surface mscorlib?  (we do not support desktop)